src/content/guides/creating-your-own-admission-controller/index.md

```diff
@@ -0,0 +1,219 @@
+
+## Generate the certificates and the CA
+
+Since the scope of this guide is not teaching how to build a PKI bundle, we have created a script, `gen_cert.sh`, in the grumpy repository which generates a CA bundle and a key pair for our grumpy server. Also we need to provide the CA in the webhook displayed above, in order to allow the Kubernetes API to create a secure connection against our shiny controller.
+
+```bash
+$ ./gen_cert.sh
+```
+
+__Note:__ Inside the aforementioned script there are comments explaining the commands executed in case you are interested in what was done under the hood.
+
+For the purpose of this tutorial, our validation webhook configuration must contain an encoded certificate authority. Besides creating the certificates and the CA, the script later injects it into the manifest used to deploy our server.
+
+```bash
+$ cat manifest.yaml | grep caBundle
+```
+
+In the next step, we need to create a secret to place the certificates. After we apply the manifest, the pod will be able to mount the secret files into a directory.
+
+```bash
+$ kubectl create secret generic grumpy \
+        --from-file=key.pem=certs/grumpy-key.pem \
+        --from-file=cert.pem=certs/grumpy-crt.pem
+```
+
+## Deploy validation controller
+
+In order to deploy the server, we will use a deployment with a single replica which mounts the certs generated to expose a secure REST endpoint where the pod request will be submitted. At the same time, we will expose the controller through the service to configure the DNS as we have defined in the webhook resource.
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: grumpy
+  namespace: default
+spec:
+  replicas: 1
+  template:
+    spec:
+      containers:
+        - name: webhook
+          image: giantswarm/grumpy:1.0.0
+          ...
+          volumeMounts:
+            - name: webhook-certs
+              mountPath: /etc/certs
+        ...
+      volumes:
+        - name: webhook-certs
+          secret:
+            secretName: grumpy
+--- 
+apiVersion: v1
+kind: Service
+metadata:
+  name: grumpy
+  namespace: default
+spec:
+  ports:
+  - name: webhook
+    port: 443
+    targetPort: 8080
+    ...
+```
+
+Applying the manifest should be enough. It also contains the webhook commented before.
+
+```bash
+$ kubectl apply -f manifest.yaml
+```
+
+Now the server should be running and ready to validate the creation of new pods.
+
+## Verify validation controller works
+
+Let's try to create a simple pod with a non-matching name.
+
+```bash
+$ kubectl apply -f non-smooth-app.yaml
+Error from server: error when creating "non-smooth-app.yaml": admission webhook "grumpy.giantswarm.org" denied the request: Keep calm and not add more crap in the cluster!
```

Comment:
you have typos in the controller, you might wanna change that to:

`Keep calm and don't add more crap to the cluster!`

Reply:
```suggestion
Error from server: error when creating "non-smooth-app.yaml": admission webhook "grumpy.giantswarm.org" denied the request: Keep calm and don't add more crap to the cluster!
```